File: backend/src/domain/repositories/IHelpOfferRepository.ts
import { HelpOffer, HelpOfferStatus } from '../entities/HelpOffer';
import { HelpRequestType } from '../entities/HelpRequest';

export interface IHelpOfferRepository {
  findById(id: string): Promise<HelpOffer | null>;
  findByEmergency(emergencyId: string): Promise<HelpOffer[]>;
  findByVolunteer(volunteerId: string): Promise<HelpOffer[]>;
  findAvailableByEmergency(emergencyId: string): Promise<HelpOffer[]>;
  findAvailableByType(emergencyId: string, type: HelpRequestType): Promise<HelpOffer[]>;
  findByStatus(emergencyId: string, status: HelpOfferStatus): Promise<HelpOffer[]>;
  save(offer: HelpOffer): Promise<HelpOffer>;
  update(offer: HelpOffer): Promise<HelpOffer>;
  delete(id: string): Promise<void>;
}

File: backend-dotnet/src/Hestia.Domain/Repositories/IHelpOfferRepository.cs
using Hestia.Domain.Entities;
using Hestia.Domain.Enums;

namespace Hestia.Domain.Repositories;

public interface IHelpOfferRepository
{
    Task<HelpOffer?> FindByIdAsync(Guid id, CancellationToken ct = default);
    Task<IReadOnlyList<HelpOffer>> FindByEmergencyAsync(Guid emergencyId, CancellationToken ct = default);
    Task<IReadOnlyList<HelpOffer>> FindByVolunteerAsync(Guid volunteerId, CancellationToken ct = default);
    Task<IReadOnlyList<HelpOffer>> FindAvailableByEmergencyAsync(Guid emergencyId, CancellationToken ct = default);
    Task<IReadOnlyList<HelpOffer>> FindAvailableByTypeAsync(Guid emergencyId, HelpType type, CancellationToken ct = default);
    Task<IReadOnlyList<HelpOffer>> FindByStatusAsync(Guid emergencyId, HelpOfferStatus status, CancellationToken ct = default);
    Task<HelpOffer> SaveAsync(HelpOffer offer, CancellationToken ct = default);
    Task<HelpOffer> UpdateAsync(HelpOffer offer, CancellationToken ct = default);
    Task DeleteAsync(Guid id, CancellationToken ct = default);
}
